src/react-components/create-object-dialog.js

```diff
@@ -22,52 +40,84 @@ export default class CreateObjectDialog extends Component {
   };
 
   componentDidMount() {
-    this.setState({ url: lastUrl }, () => {
-      this.onUrlChange({ target: this.input });
-    });
+    this.setState({ url: lastUrl });
   }
 
   componentWillUnmount() {
     lastUrl = this.state.url;
   }
 
   onUrlChange = e => {
-    if (e && e.target.value && e.target.value !== "") {
+    let url;
+    try {
+      url = new URL(e.target.value);
+    } catch (_) {
       this.setState({
-        url: e.target.value,
-        attributionImage: e.target.validity.valid && attributionHostnames[new URL(e.target.value).hostname]
+        url: e.target && e.target.value
+        // Not a valid URL, so don't try to set attributeImage using url.hostname
       });
+      return;
     }
+    this.setState({
+      url: e.target && e.target.value,
+      attributionImage: e.target && e.target.value && e.target.validity.valid && attributionHostnames[url.hostname]
+    });
+  };
+
+  onFileChange = e => {
+    this.setState({
+      file: e.target.files[0],
+      fileName: e.target.files[0].name
+    });
   };
 
   onCreateClicked = () => {
-    this.props.onCreateObject(this.state.url || DEFAULT_OBJECT_URL);
+    this.props.onCreateObject(this.state.file || this.state.url || DEFAULT_OBJECT_URL);
     this.props.onCloseDialog();
   };
 
+  reset = e => {
+    e.preventDefault();
+    this.setState({
+      url: "",
+      file: null,
+      fileName: ""
+    });
+  };
+
   render() {
+    const cancelButton = (
+      <label className={cx(styles.smallButton, styles.cancelIcon)} onClick={this.reset}>
+        <FontAwesomeIcon icon={faTimes} />
+      </label>
+    );
+    const uploadButton = (
+      <label htmlFor={fileInputId} className={cx(styles.smallButton, styles.uploadIcon)}>
+        <FontAwesomeIcon icon={faPaperclip} />
+      </label>
+    );
+    const filenameLabel = (
+      <label className={cx(styles.fileNameLabel, "add-media-form__link_field")}>{this.state.fileName}</label>
```

Comment:
I assume these non local classes are just a carryover? Seems like since they are classes specific to this we should just copy them over into `create-object-dialog.scss`